src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -248,101 +212,80 @@ private Task ProcessAuthentication(bool isAsync = false, bool isApm = false, Can
         //
         // This is used to reply on re-handshake when received SEC_I_RENEGOTIATE on Read().
         //
-        private async Task ReplyOnReAuthenticationAsync(byte[] buffer, CancellationToken cancellationToken)
+        private async Task ReplyOnReAuthenticationAsync<TReadAdapter>(TReadAdapter adapter, byte[] buffer)
```

Comment:
Nit: Now that we've combined the read/write adapters, can we rename `TReadAdapter` to something that's not read specific, e.g. just `TAdapter` or `TIOAdapter` or something like that?